src/components/pill-selector-dropdown/waitForInputFilter.js

```diff
@@ -0,0 +1,15 @@
+// @flow
+import defaultDropdownFilter from './defaultDropdownFilter';
+
+import type { Option } from './flowTypes';
+
+// Same as default filter except that it won't show the selector dropdown until the user begins typing
+// (as opposed to showing it when the input is focused)
+function waitForInputFilter(options: Array<Option>, selectedValues?: Array<Option>, inputText?: string): Array<Option> {
```

Comment:
I am wondering if this should be part of the parent app than here. Since I am guessing you will pass this filter as a prop for the Formik `<Field>`.

On the other hand, we could create a sub-folder called `filters` and move both this and default one in there and then other filters can also be added.

A 3rd way would be that maybe we have this a a prop something like `is(should?)DropdownOpenOnFocus` and then just use that boolean within PillSelectorDropdownField to return [] if true OR do ->

Additively 4th option would be that we pass the prop from the above as a new prop to PillSelectorDropdown which passes the prop to SelectorDropdown.js and guard the handleFocus() function which is what opens it. So selectordropdown gets a new prop to prevent opening on focus like you added the prop to focus 1st item.

Reply:
@priyajeet do you have a preference for any of these options? I'm leaning towards the 4th since, now that I think about it, this behavior is not directly related to filtering. It probably makes more sense to handle this with a prop that is independent of filtering.

Reply:
4th one would have been my pref too

Reply:
@priyajeet on close inspection it looks like this is related to filtering after all since in general we don't want to display the dropdown when the filter value is empty. So for example we also want to close the dropdown if the user clears the input after typing, but the `SelectorDropdown` doesn't have context on that.

Are you ok with using the second option of having a "filters" folder? I think it would be good to keep the filter in this project since it could be useful to other people.

Reply:
thats works